ote/src/cljs/ote/app/controller/transport_operator.cljs

```diff
@@ -453,18 +453,21 @@
 
   EnsureUniqueBusinessId
   (process-event [{business-id :business-id} app]
-    (let [business-id-validation-error (validation/validate-rule :business-id nil business-id)]
-      (when (nil? business-id-validation-error)
-        (comm/get! (str "transport-operator/ensure-unique-business-id/" business-id)
-                   {:on-success (send-async! ->EnsureUniqueBusinessIdResponse)}))
+    (let [bid-validation-error (validation/validate-rule :business-id nil business-id)
+          resolve-bid-uniqueness-state (fn [state bid-validation-error]
+                                         (if (nil? bid-validation-error)
+                                           (do
+                                             (comm/get! (str "transport-operator/ensure-unique-business-id/" business-id)
```

Comment:
Looks scary that there are side effects in the right side of a let binding, might be easy to miss?

Reply:
The reason is structuring code in readable blocks.
Conder that a func call requires n args, but those are dynamic and calculated. Code to resolve the args can be embedded to the func call, or alternatively resolved earlier and bound to a value.
It is not possible to describe complex logic in a maintainable way without this kind of structure - although alternative suggestions are always welcome :)